### PR TITLE
Add support for regex operators.

### PR DESCRIPTION
**Pull request by jenkoian**
Adding support for operators listed here: https://www.postgresql.org/docs/9.3/functions-matching.html#FUNCTIONS-POSIX-REGEXP

### CODE REVIEW

File: tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/NotIRegexpTest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+use MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\NotIRegexp;
+use Tests\MartinGeorgiev\Doctrine\Fixtures\Entity\ContainsTexts;
+
+class NotIRegexpTest extends TestCase
+{
+    protected function getStringFunctions(): array
+    {
+        return [
+            'NOTIREGEXP' => NotIRegexp::class,
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/NotIRegexpTest.php:15]:
```suggestion
            'NOT_IREGEXP' => NotIRegexp::class,
```

File: tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/NotIRegexpTest.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        return [
+            "SELECT (c0_.text1 !~* '.*Thomas.*') AS sclr_0 FROM ContainsTexts c0_",
+        ];
+    }
+
+    protected function getDqlStatements(): array
+    {
+        return [
+            \sprintf("SELECT NOTIREGEXP(e.text1, '.*Thomas.*') FROM %s e", ContainsTexts::class),
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/NotIRegexpTest.php:29]:
```suggestion
            \sprintf("SELECT NOT_IREGEXP(e.text1, '.*Thomas.*') FROM %s e", ContainsTexts::class),
```

File: tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/NotRegexpTest.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+use MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\NotRegexp;
+use Tests\MartinGeorgiev\Doctrine\Fixtures\Entity\ContainsTexts;
+
+class NotRegexpTest extends TestCase
+{
+    protected function getStringFunctions(): array
+    {
+        return [
+            'NOTREGEXP' => NotRegexp::class,
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/NotRegexpTest.php:15]:
```suggestion
            'NOT_REGEXP' => NotRegexp::class,
```

File: tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/NotRegexpTest.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        return [
+            "SELECT (c0_.text1 !~ '.*thomas.*') AS sclr_0 FROM ContainsTexts c0_",
+        ];
+    }
+
+    protected function getDqlStatements(): array
+    {
+        return [
+            \sprintf("SELECT NOTREGEXP(e.text1, '.*thomas.*') FROM %s e", ContainsTexts::class),
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/NotRegexpTest.php:29]:
```suggestion
            \sprintf("SELECT NOT_REGEXP(e.text1, '.*thomas.*') FROM %s e", ContainsTexts::class),
```

**Comment by martin-georgiev** [comment]
Thank you for your contribution [jenkoian](https://github.com/jenkoian) 🍻 !

It looks awesome - I have added some rather nitpicking comments, let me know your thoughts about them.

**Comment by jenkoian** [comment]
Sorry for the delay, all suggested changes now made. I've pushed it as a new commit to make reviewing easier, let me know if you want me to squash 👍

**Review by martin-georgiev — APPROVED** [review]

**Pull request merged**